### PR TITLE
fix: update kustomize to 4.0.5

### PR DESCRIPTION
**Pull request by sbose78**
Note, v4.0.5 is also the candidate for kubectl integration

---

Signed-off-by: Shoubhik Bose <shbose@redhat.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by sbose78** [comment]
yet to test..

**Comment by r0bj** [comment]
Just FYI, kustomize 4.x has unresolved bug https://github.com/kubernetes-sigs/kustomize/issues/3446 - using integer key generates error.

**Comment by sbose78** [comment]
Thank you!

File: hack/tool-versions.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ jq_version=1.6
 ksonnet_version=0.13.1
 kubectl_version=1.17.8
 kubectx_version=0.6.3
-kustomize3_version=3.9.4
+kustomize3_version=4.0.5
```
Comment (cguertin14) [inline hack/tool-versions.sh:10]: Probably will need to add a `kustomize4_version` here instead.

**Review by cguertin14 — CHANGES_REQUESTED** [review]

**Comment by alexmt** [comment]
Closing in favor of https://github.com/argoproj/argo-cd/pull/6059

**Pull request closed without merging**